### PR TITLE
[ticket/15135] Parse language files in acp_extensions instead of md manager

### PR DESCRIPTION
**Pull request by marc1706**
PHPBB3-15135

Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [ ] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15135

### CODE REVIEW

File: phpBB/includes/acp/acp_extensions.php [hunk 1]
```diff
@@ -162,7 +162,8 @@ function main()
 				}
 				catch (\phpbb\extension\exception $e)
 				{
-					trigger_error($e . adm_back_link($this->u_action), E_USER_WARNING);
+					$message = call_user_func_array(array($this->user, 'lang'), array_merge(array($e->getMessage()), $e->get_parameters()));
```
Comment (Nicofuma) [inline phpBB/includes/acp/acp_extensions.php]: No more back link? (To be honest I never know when to add them and when not to...)

Comment (marc1706) [inline phpBB/includes/acp/acp_extensions.php]: Oh, heh. Might actually be nice to have it still. Will add it back.

**Comment by javiexin** [comment]
Apologies, as i see this was caused by one of my PRs.

File: phpBB/includes/acp/acp_extensions.php (hunk 1 above)
Comment (javiexin) [inline phpBB/includes/acp/acp_extensions.php]: Same change is required in line 196

File: phpBB/phpbb/extension/metadata_manager.php [hunk 2]
```diff
@@ -311,7 +311,7 @@ public function validate_require_php()
 	{
 		if (!isset($this->metadata['require']['php']))
 		{
-			throw new \phpbb\extension\exception($this->user->lang('META_FIELD_NOT_SET', 'require php'));
+			throw new \phpbb\extension\exception('META_FIELD_NOT_SET', 'require php');
```
Comment (javiexin) [inline phpBB/phpbb/extension/metadata_manager.php]: Likewise `array('require php')`?

File: phpBB/phpbb/extension/metadata_manager.php [hunk 3]
```diff
@@ -295,7 +295,7 @@ public function validate_require_phpbb()
 	{
 		if (!isset($this->metadata['extra']['soft-require']['phpbb/phpbb']))
 		{
-			throw new \phpbb\extension\exception($this->user->lang('META_FIELD_NOT_SET', 'soft-require'));
+			throw new \phpbb\extension\exception('META_FIELD_NOT_SET', 'soft-require');
```
Comment (javiexin) [inline phpBB/phpbb/extension/metadata_manager.php]: Shouldn't this be `array('soft-require')`?

**Pull request merged**